### PR TITLE
page_store: define write_buffer API

### LINKED ISSUE #12: Define WriteBuffer module API

### PR DESCRIPTION
**Pull request by w41ter**
This PR closes #12.

### CODE REVIEW

**Comment by w41ter** [comment]
> maybe we need rename page_id to page_addr :)

fixed

File: src/page_store/write_buffer.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+bitflags! {
+    struct RecordFlags: u32 {
+        const NORMAL_PAGE   = 0b0000_0001;
+        const DELETED_PAGES = 0b0000_0010;
+
+        const TOMBSTONE     = 0b1000_0000;
+    }
+}
+
+const _: () = {
```
Comment (huachaohuang) [inline src/page_store/write_buffer.rs:227]: What's this for?

Comment (huachaohuang) [reply src/page_store/write_buffer.rs:227]: I think the compiler will handle it. It is obvious enough if you implement that trait for the type.

**Comment by huachaohuang** [comment]
Please resolve conflicts.

**Pull request merged**